docs/DeveloperGuide.md

```diff
@@ -1,34 +1,614 @@
+The converter module consists of one class which converts the format of date and time using a custom format defined by
+special formatting characters (ie. SimpleDateFormat). This class is primarily used to format a user-input date and time
+in the `Appointment` class. As illustrated below, its methods are called during the creation of the `Appointment`
+object constructor. 
+
+![](images/SD_Converter.png)
+
+###### [Back to top](#table-of-content)
+
+#### 2.2.3 BRANDON storage module
+
+The Storage module consists of 3 different classes. 
+The PatientList and AppointmentList classes act as data structures to store the records of Patient and Appointment 
+objects respectively. They function as ADTs, where various commands from Command objects can manipulate the records within.
+The Storage class manages the load and save operations involving the PatientList and PatientList class. 
+These operations are usually invoked on startup, whenever changes are made to the ADTs and before exiting the program.
+The class diagram for the storage module is as seen below: 
+
+
+![](images/storageclass.PNG)
+
+&nbsp;
+
+On startup, Duke invokes the loadSavedAppointment() and loadSavedPatient() methods in Storage. This allows the program 
+to retrieve previously stored data from a .txt file and convert it into the static AppointmentList and PatientList objects for use
+within the program. 
+
+The Storage object creates a Scanner object that will parse individual lines in the .txt file, convert them into
+new Appointments, and then add them to an ArrayList called `appointmentListToReturn`. This `appointmentListToReturn` will be passed back to Duke to
+construct the static AppointmentList. The sequence diagram is shown below:
+
+![](images/loadsavedappt_seq.PNG)
+![](images/loadsavedappt_ref1.PNG)
+
+When the static AppointmentList or PatientList has changes, or the program is exiting, saveAppointmentList() or savePatientList() 
+is invoked respectively. This allows the Storage object to back up existing records to a local .txt file.
+
+The Storage object will create a FileWriter object called `fw`. The command will then iterate through the existing AppointmentList
+and parse each Appointment within, converting it to a string. `fw` then writes this string to the .txt file.
+The sequence diagram is shown below:
+
+![](images/saveapptlist_seq.PNG)
+
+###### [Back to top](#table-of-content)
+
+#### 2.2.4 A&D command module 
+
+The command module consist of 11 different classes, where each class does a different command by itself. 
+These classes allows the patients and appointments to be added into HAMS, allows the updating of patient and 
+appointment details, allows the admin to delete patients and appointment information and allows of listing of the 
+different patients and the appointments in HAMS. In addition, the classes also deals with displaying the list of 
+commands and as well as to allowing HAMS to exit.
+
+![](images/Command%20class%20diagram.png)
```

Comment:
![image](https://user-images.githubusercontent.com/31772358/77993613-0f0dc400-735b-11ea-8b84-d30d60092f78.png)

Consider omitting some details (e.g. private variables) to make the diagram less complicated?